include/roboteam_ai/skills/AvoidBall.h

```diff
@@ -21,7 +21,7 @@ class AvoidBall : public Skill {
 
     public:
         explicit AvoidBall(std::string name, bt::Blackboard::Ptr blackboard = nullptr);
-        void onInitialize() override;
+        void onInitialize(std::string type);
```

Comment:
Ooh, I like this way of doing it.

Reply:
isn't onInitialize() literally a constructor? By definition a constructor is called when you initialize.